S17-lowlevel/lock.t

```diff
@@ -100,9 +100,7 @@ plan 23;
         my $counter = 0;
         my $t1 = Thread.start({
             $l.protect({
-                while $counter == 0 {
```

Comment:
But we haven't removed this functionality, only added new. So, perhaps we need a new test?

Reply:
That's tested a bit down the line.

Reply:
Yes, but it's a bit different scenario down there. Because `wait` is a multi, there is very tiny, but still a non-zero, chance that something could break. For example, re-entering the method would break whereas the predicate version doesn't re-enter and would be immune to the problem.

Reply:
The main problem I see here is that this code is actually written to *not* run. So we're not actually testing it. Let me see what can be done.

Reply:
`.wait()` is also tested in line 68 in the same file. Is that what you mean? That's what I meant in the PR body. There were two instances of wait(); I've turned one of them to test the other multi.

Reply:
Actually, I meant line 150, but that's irrelevant. My point is that neither 68 nor 150 are testing for repetitive invocations of `wait` withing a lock. And your PR actually removes the only case where the case is tested.